chore: Add support for Github Actions

rynmrtn opened the pull request:
- Github Actions yaml for test + lint
- README w/ go-doc reference link
- doc.go for go docs
- go mod init

rodrigopv commented on networks/ip.go:
```diff
@@ -0,0 +1,6 @@
+
+
+type IPAddress struct {
+ value string,
```
This should raise a golangci error, making the action to fail, will remove once we assert that the CI fails

rynmrtn replied:
i would rather remove this for this PR and push it up as a separate one that executes github actions

rodrigopv commented on networks/ip_test.go:
```diff
@@ -0,0 +1,16 @@
+package networks
```
This file will also be removed after verifying that the CI fails after running tests

rynmrtn replied:
i would rather remove this for this PR and push it up as a separate one that executes github actions

rynmrtn committed:
chore: Add support for Github Actions

- Github Actions yaml for test + lint
- README w/ go-doc reference link
- doc.go for go docs
- go mod init
